### DISCUSSION TITLE
Get user information in a custom route

### DISCUSSION THREAD

**Discussion by shubhendumadhukar** [pocketbase/pocketbase#1911] [Q&A] [answered]
I am trying to build a custom route per the example shown for `OnBeforeServe` hook. I was able to create a route and I replaced the `RequireAdminAuth()` with `RequireRecordAuth()`, so that any authorised user can call this endpoint. Next I have extracted the token `authToken := c.Request().Header.Get("Authorization")`.

This next part, not necessarily pocketbase related, but it is something I am struggling with and I was wondering if there is a simpler way of doing it currently (maybe echo context has user information?). Anyway, I would like to extract the id string from authToken and then subsequently use it to fetch user record and perform some other validations. But I don't seem to be able to extract id from the jwt payload. I did stumble upon this example (https://pkg.go.dev/github.com/golang-jwt/jwt/v5#example-Parse-Hmac) but couldn't get it work. It would be great if someone could point me in the right direction.

**Comment by ganigeorgiev** [comment, marked as answer]
If you want to get the authenticated record or admin id, you can check https://pocketbase.io/docs/router/#get-logged-admin-or-app-user-in-a-route-handler:

```go
// to get the authenticated admin
admin, _ := c.Get(apis.ContextAdminKey).(*models.Admin)

// or to get the authenticated record
authRecord, _ := c.Get(apis.ContextAuthRecordKey).(*models.Record)
```

**Reply by shubhendumadhukar** [reply]
That is exactly what I am looking for. Thank you!

**Reply by shubhendumadhukar** [reply]
I have a follow up question, I tried searching it in the docs but didn't have any luck...how do I get the authRecord inside a hook?

**Reply by ganigeorgiev** [reply]
You can access the current request auth record (if any) only from the `*Request` hooks via the `e.HttpContext` field.

You can find an example in https://pocketbase.io/docs/router/#get-logged-admin-or-app-user-in-a-request-hook.
